js/ui/action_button/action_button_base.js

```diff
@@ -0,0 +1,223 @@
+
+            action._options.position = {
+                of: this.content(),
+                at: "center",
+                my: "center",
+                offset: {
+                    x: 0,
+                    y: -actionOffsetY
+                }
+            };
+
+            action._options.animation.show.delay = action._options.animation.show.duration * i;
+            action._options.animation.hide.delay = action._options.animation.hide.duration * (lastActionIndex - i);
+
+            this._actionItems.push(this._createComponent($actionElement, ActionButtonItem, action._options));
+        }
+    },
+
+    _optionChanged(args) {
+        switch(args.name) {
+            case "actions":
+                this._renderActions();
+                break;
+            case "position":
+                this._renderActions();
+                break;
+            case "maxActionButtonCount":
+                this._renderActions();
+                break;
+            case "icon":
+                this._renderIcon();
+                break;
+            case "closeIcon":
+                this._renderCloseIcon();
+                break;
+            default:
+                this.callBase(args);
+        }
+    }
+});
+
+exports.init = function(newAction) {
+    let isActionExist = false;
+    if(!actionButtonBase) {
+        const $fabMainElement = $("<div>")
+            .appendTo(swatch.getSwatchContainer(newAction.$element()));
+        actionButtonBase = new ActionButtonBase($fabMainElement, {
+            icon: newAction._options.icon || config().actionButtonConfig.icon,
+            onClick: newAction._options.onClick,
+            actions: [ newAction ]
+        });
+    } else {
+        const savedActions = actionButtonBase.option("actions");
+
+        savedActions.forEach(action => {
+            if(action._options.id === newAction._options.id) {
+                isActionExist = true;
+                return newAction;
+            }
+        });
+
+        if(!isActionExist) {
+            savedActions.push(newAction);
+            actionButtonBase.option("actions", savedActions);
+            actionButtonBase.option("icon", config().actionButtonConfig.icon);
+            actionButtonBase.option("onClick", null);
+        } else if(savedActions.length === 1) {
+            actionButtonBase.option("icon", newAction._options.icon);
+            actionButtonBase.option("onClick", newAction._options.onClick);
+        } else {
+            actionButtonBase.option("actions", savedActions);
+        }
+    }
+};
+
+exports.dispose = function(actionId) {
+    let savedActions = actionButtonBase.option("actions");
+    if(savedActions.length === 1) {
+        actionButtonBase.option("actions", []);
+        actionButtonBase.dispose();
+        actionButtonBase.element().remove();
+        actionButtonBase = null;
+    } else {
+        savedActions = savedActions.filter(action => {
```

Comment:
I think `dispose` should dispose the widget completely in both cases.

Reply:
Ok. I understand. `dispose` should dispose the widget completely in any case. We need method like disposeAction(actionId) that will call the `dispose` method if it was the single action.